apps/server/src/modules/room/api/mapper/room.mapper.ts

```diff
@@ -1,25 +1,45 @@
 import { Page } from '@shared/domain/domainobject';
-import { RoomPaginationParams } from '../dto/request/room-pagination.params';
-import { RoomResponse, RoomListResponse } from '../dto';
 import { Room } from '../../domain/do/room.do';
+import { RoomPaginationParams } from '../dto/request/room-pagination.params';
+import { RoomDetailsResponse } from '../dto/response/room-details.response';
+import { RoomItemResponse } from '../dto/response/room-item.response';
+import { RoomListResponse } from '../dto/response/room-list.response';
 
 export class RoomMapper {
-	static mapToRoomResponse(room: Room): RoomResponse {
-		const response = new RoomResponse({
+	static mapToRoomResponse(room: Room): RoomItemResponse {
+		const response = new RoomItemResponse({
 			id: room.id,
 			name: room.name,
 			color: room.color,
 			startDate: room.startDate,
-			untilDate: room.untilDate,
+			endDate: room.endDate,
+			createdAt: room.createdAt,
+			updatedAt: room.updatedAt,
 		});
 
 		return response;
 	}
 
 	static mapToRoomListResponse(rooms: Page<Room>, pagination: RoomPaginationParams): RoomListResponse {
-		const roomResponseData: RoomResponse[] = rooms.data.map((room): RoomResponse => this.mapToRoomResponse(room));
+		const roomResponseData: RoomItemResponse[] = rooms.data.map(
+			(room): RoomItemResponse => this.mapToRoomResponse(room)
+		);
 		const response = new RoomListResponse(roomResponseData, rooms.total, pagination.skip, pagination.limit);
 
 		return response;
 	}
+
+	static mapToRoomDetailsResponse(room: Room): RoomDetailsResponse {
```

Comment:
looks the same as `mapToRoomResponse`

Maybe just use one?

Reply:
The method `mapToRoomItemResponse` had return type `RoomDetailsResponse`. This was wrong. Changed it to `RoomItemResponse`.
I think it is a good idea to have different DTOs for the list item and the details because of the different use cases. We might have a lot more information in the details in the future than in the item.